File: typescript/src/workshop/m5/s09_overloading/step2/PriceList.ts
import type { Money } from '../../../shared/Money.js';
import type { Ticket } from './Ticket.js';

/** Krok 2: bez zmian. */
export class PriceList {
  price(ticket: Ticket): Money {
    return ticket.basePrice().minus(ticket.basePrice().percent(ticket.discountPercent()));
  }
}

File: csharp/src/Training.Workshop/M5/S09Overloading/Step2/PriceList.cs
using Training.Workshop.Shared;

namespace Training.Workshop.M5.S09Overloading.Step2;

/// <summary>Krok 2: bez zmian.</summary>
public sealed class PriceList
{
    public Money Price(Ticket ticket)
    {
        return ticket.BasePrice.Minus(ticket.BasePrice.Percent(ticket.DiscountPercent));
    }
}
